modules/aug/src/transforms.cpp

```diff
@@ -0,0 +1,585 @@
+
+        // NOTE: cv::randomCrop or randomCrop?
+        void randomCrop(InputArray _src, OutputArray _dst, const Size& sz, const Vec4i& padding, bool pad_if_need, int fill, int padding_mode){
+            // FIXME: whether the size of src should be (src.cols+left+right, src.rows+top+bottom)
+
+            Mat src = _src.getMat();
+
+            if(padding != Vec4i()){
+                copyMakeBorder(src, src, padding[0], padding[1], padding[2], padding[3], padding_mode, fill);
+            }
+
+            // NOTE: make sure src.rows == src.size().height and src.cols = src.size().width
+            // pad the height if needed
+            if(pad_if_need && src.rows < sz.height){
+                Vec4i _padding = {sz.height - src.rows, sz.height - src.rows, 0, 0};
+                copyMakeBorder(src, src, _padding[0], _padding[1], _padding[2], _padding[3], padding_mode, fill);
+            }
+            // pad the width if needed
+            if(pad_if_need && src.cols < sz.width){
+                Vec4i _padding = {0, 0, sz.width - src.cols, sz.width - src.cols};
+                copyMakeBorder(src, src, _padding[0], _padding[1], _padding[2], _padding[3], padding_mode, fill);
+            }
+
+            int x, y;
+            getRandomCropParams(src.rows, src.cols, sz.height, sz.width, &x, &y);
+            // FIXME: for test only
+            std::cout << x << " " << y << std::endl;
+            Mat RoI(src, Rect(x, y, sz.width, sz.height));
+            RoI.copyTo(_dst);
+            // inplace operation
+            // NOTE: inplace operation not works in converting from python to numpy
+//        _dst.move(RoI);
+        }
+
+//    CV_EXPORTS_W void randomCropV1(InputOutputArray _src, const Size& sz, const Vec4i& padding, bool pad_if_need, int fill, int padding_mode){
+//        Mat src = _src.getMat();
+//
+//        if(padding != Vec4i()){
+//            copyMakeBorder(src, src, padding[0], padding[1], padding[2], padding[3], padding_mode, fill);
+//        }
+//
+//        // NOTE: make sure src.rows == src.size().height and src.cols = src.size().width
+//        // pad the height if needed
+//        if(pad_if_need && src.rows < sz.height){
+//            Vec4i _padding = {sz.height - src.rows, sz.height - src.rows, 0, 0};
+//            copyMakeBorder(src, src, _padding[0], _padding[1], _padding[2], _padding[3], padding_mode, fill);
+//        }
+//        // pad the width if needed
+//        if(pad_if_need && src.cols < sz.width){
+//            Vec4i _padding = {0, 0, sz.width - src.cols, sz.width - src.cols};
+//            copyMakeBorder(src, src, _padding[0], _padding[1], _padding[2], _padding[3], padding_mode, fill);
+//        }
+//
+//        int x, y;
+//        getRandomCropParams(src.rows, src.cols, sz.height, sz.width, &x, &y);
+//        Mat cropped(src, Rect(x, y, sz.width, sz.height));
+//        (*(Mat*)_src.getObj()) = cropped;
+//    }
```

Comment:
Please remove code which are not finished, experimental, for debug and the like.